types/three/src/nodes/tsl/TSLCore.d.ts

```diff
@@ -234,7 +234,11 @@ export function nodeImmutable<T>(
 interface Layout {
     name: string;
     type: string;
-    inputs: { name: string; type: string }[];
+    inputs: {
+        name: string;
+        type: "bool" | "int" | "float" | "vec2" | "vec3" | "vec4";
```

Comment:
Aren't there more types than this?

Reply:
> Aren't there more possible types than this?

I don't remember why I wrote that...

Maybe use the type in the 'NodeUtils.js -> getValueFromType' method, which should be more complete

```ts
type: "bool" | "int" | "uint" | "float" | "vec2" | "vec3" | "vec4" | "mat2" | "mat3" | "mat4" | "color" | "string" | "ArrayBuffer";
```

Reply:
I'm just going to change it back to `string` for now until I have time to look into it some more.